src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final ColorSupplier COLOR_SUPPLIER = new ColorSupplier();
```

Comment:
Each FigureSupplier should have its own ColorSupplier, it should not be a constant